src/webextension/background/accounts/index.js

```diff
@@ -228,6 +187,118 @@ export class Account {
       avatar: this.avatar,
     };
   }
+
+  async token() {
+    // always return null if user is GUEST
+    if (this.mode === GUEST) {
+      // XXXX: use DataStoreError
+      throw new Error("AUTH: requires FxA");
+    }
+
+    // check if token present / unexpired / valid
+    let info = await this.updateUserInfo();
+    if (!info || !info.access_token) {
+      // refresh
+      await this.updateAccessToken();
+      info = await this.updateUserInfo();
+    }
+
+    if (!info || !info.access_token) {
+      // XXXX: use DataStoreError
+      throw new Error("AUTH: no access token");
+    }
+
+    return info.access_token;
+  }
+
+
+  async updateAccessToken(params, appKey) {
+    const cfg = configs[this.config];
+    let info = this.info || {};
+
+    if (!params) {
+      // assume "refresh_token" exchange
+      if (!info.refresh_token) {
+        // XXXX: use DataStoreError
+        throw new Error("AUTH: no refresh token");
+      }
+
+      params = {
+        grant_type: "refresh_token",
+        refresh_token: info.refresh_token,
+        client_id: cfg.client_id,
+      };
+    }
+
+    const request = {
+      method: "post",
+      headers: {
+        "content-type": "application/json",
+      },
+      cache: "no-cache",
+      body: JSON.stringify(params),
+    };
+    const oauthInfo = await fetchFromEndPoint("token", cfg.token_endpoint, request);
+    let keys = info.keys || new Map();
+    if (oauthInfo.keys_jwe && appKey) {
+      // forget previous keys before decrypting new bundle
+      keys.clear();
+
+      let bundle = await jose.JWE.createDecrypt(appKey).decrypt(oauthInfo.keys_jwe);
+      bundle = JSON.parse(new TextDecoder().decode(bundle.payload));
+      const pending = Object.keys(bundle).map(async (name) => {
+        let key = bundle[name];
+        key = await jose.JWK.asKey(key);
+        keys.set(name, key);
+      });
+      await Promise.all(pending);
+    }
+
+    this.info = info = {
+      ...info,
+      access_token: oauthInfo.access_token,
+      expires_at: 0 | (Date.now() / 1000) + oauthInfo.expires_in,
```

Comment:
This `0 | ` looks new; you're truncating to an int right? Could we use `Math.trunc` or `Math.floor` instead? Those would be clearer (`Math.trunc` should have identical behavior, but `Math.floor` is probably clearer; it only matters if the arg is negative, which it shouldn't be). If there's a particular reason those wouldn't work, then feel free to leave it the way it is...

Reply:
it is for int truncation, and the habit I got into was using this over `Math.floor` because of a very significant (order of magnitude) performance difference in the past.  I can switch them to floor, though.